poh/src/poh_recorder.rs

```diff
@@ -452,11 +452,8 @@ impl PohRecorder {
         })
     }
 
-    fn prev_slot_was_mine(&self, my_pubkey: &Pubkey, current_slot: Slot) -> bool {
-        if let Some(leader_id) = self
-            .leader_schedule_cache
-            .slot_leader_at(current_slot.saturating_sub(1), None)
-        {
+    fn prev_slot_was_mine(&self, my_pubkey: &Pubkey, parent_slot: Slot) -> bool {
+        if let Some(leader_id) = self.leader_schedule_cache.slot_leader_at(parent_slot, None) {
```

Comment:
Oh and we can skip this cache lookup because we can get the start bank's leader directly from `start_bank.collector_id()`